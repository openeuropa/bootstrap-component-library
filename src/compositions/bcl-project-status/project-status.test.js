import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
import demoOngoing from "@openeuropa/bcl-project-status/data/data--ongoing";
import demoPlanned from "@openeuropa/bcl-project-status/data/data--planned";
import demoClosed from "@openeuropa/bcl-project-status/data/data--closed";
import demoContribs from "@openeuropa/bcl-project-status/data/data--contributions.js";

describe("OE - Project status", () => {
  let template = "@oe-bcl/bcl-project-status/project-status.html.twig";
  let render = (params) => renderTwigFileAsNode(template, params, true);

  test(`ongoing renders correctly`, () => {
    expect.assertions(1);

    return expect(render(demoOngoing)).resolves.toMatchSnapshot();
  });

  test(`planned renders correctly`, () => {
    expect.assertions(1);

    return expect(render(demoPlanned)).resolves.toMatchSnapshot();
  });

  test(`closed renders correctly`, () => {
    expect.assertions(1);

    return expect(render(demoClosed)).resolves.toMatchSnapshot();
  });
});

describe("OE - Project contributions", () => {
  let template = "@oe-bcl/bcl-project-status/project-contributions.html.twig";
  let render = (params) => renderTwigFileAsNode(template, params, true);

  test(`renders correctly`, () => {
    expect.assertions(1);

    return expect(render(demoContribs)).resolves.toMatchSnapshot();
  });

  test(`without chart renders correctly`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoContribs, chart: {} })
    ).resolves.toMatchSnapshot();
  });
});
