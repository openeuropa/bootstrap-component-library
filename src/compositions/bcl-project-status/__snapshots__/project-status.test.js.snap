// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Project contributions renders correctly 1`] = `
<jest>
  <div
    class="bcl-project-contributions border-bottom pb-4 mb-4"
  >
    <div
      class="row"
    >
      <div
        class="bcl-project-contributions-chart col-5 col-md-3"
      >
        <div
          class="circular-progress"
          data-percentage="70"
        >
          <span
            class="circular-progress-left"
          >
            <span
              class="circular-progress-bar"
            />
          </span>
          <span
            class="circular-progress-right"
          >
            <span
              class="circular-progress-bar"
            />
          </span>
        </div>
      </div>
      <div
        class="bcl-project-contributions-legend col-7 col-md-9 align-self-center"
      >
        <dl
          class="mb-0 d-md-grid grid-3-9"
        >
          <dt>
            <div
              class="d-flex align-items-center"
            >
              <span
                class="p-2 me-2 d-inline-block rounded-1 bg-gray-400"
              />
              Total budget
            </div>
          </dt>
          <dd>
            EUR 16.314.672,00
          </dd>
          <dt>
            <div
              class="d-flex align-items-center"
            >
              <span
                class="p-2 me-2 d-inline-block rounded-1 bg-primary"
              />
              EC contribution
            </div>
          </dt>
          <dd>
            EUR 9.994.672,00
          </dd>
        </dl>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - Project contributions without chart renders correctly 1`] = `
<jest>
  <div
    class="bcl-project-contributions border-bottom pb-4 mb-4"
  >
    <div
      class="row"
    >
      <div
        class="bcl-project-contributions-chart col-5 col-md-3"
      >
        <div
          class="circular-progress"
          data-percentage="70"
        >
          <span
            class="circular-progress-left"
          >
            <span
              class="circular-progress-bar"
            />
          </span>
          <span
            class="circular-progress-right"
          >
            <span
              class="circular-progress-bar"
            />
          </span>
        </div>
      </div>
      <div
        class="bcl-project-contributions-legend col-7 col-md-9 align-self-center"
      >
        <dl
          class="mb-0 d-md-grid grid-3-9"
        >
          <dt>
            <div
              class="d-flex align-items-center"
            >
              <span
                class="p-2 me-2 d-inline-block rounded-1 bg-gray-400"
              />
              Total budget
            </div>
          </dt>
          <dd>
            EUR 16.314.672,00
          </dd>
          <dt>
            <div
              class="d-flex align-items-center"
            >
              <span
                class="p-2 me-2 d-inline-block rounded-1 bg-primary"
              />
              EC contribution
            </div>
          </dt>
          <dd>
            EUR 9.994.672,00
          </dd>
        </dl>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - Project status closed renders correctly 1`] = `
<jest>
  <div
    class="bcl-project-status border-bottom pb-4 mb-4"
  >
    <div
      class="row"
    >
      <div
        class="col-12 col-md-3"
      >
        <p
          class="fw-bold"
        >
          Status
        </p>
        <span
          class="mb-3 mb-md-0 badge bg-dark"
        >
          Closed
        </span>
      </div>
      <div
        class="col-12 col-md-9"
      >
        <div
          class="d-flex justify-content-between"
        >
          <p
            class="fw-bold"
          >
            Start
            <br />
            <span
              class="fw-normal"
            >
              19 December 2007
            </span>
          </p>
          <p
            class="fw-bold text-end"
          >
            End
            <br />
            <span
              class="fw-normal"
            >
              19 December 2017
            </span>
          </p>
        </div>
        <div
          class="progress"
        >
          <div
            aria-label="100"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="100"
            class="progress-bar bg-dark"
            role="progressbar"
            style="width: 100%"
          />
        </div>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - Project status ongoing renders correctly 1`] = `
<jest>
  <div
    class="bcl-project-status border-bottom pb-4 mb-4"
  >
    <div
      class="row"
    >
      <div
        class="col-12 col-md-3"
      >
        <p
          class="fw-bold"
        >
          Status
        </p>
        <span
          class="mb-3 mb-md-0 badge bg-info"
        >
          Ongoing
        </span>
      </div>
      <div
        class="col-12 col-md-9"
      >
        <div
          class="d-flex justify-content-between"
        >
          <p
            class="fw-bold"
          >
            Start
            <br />
            <span
              class="fw-normal"
            >
              19 December 2020
            </span>
          </p>
          <p
            class="fw-bold text-end"
          >
            End
            <br />
            <span
              class="fw-normal"
            >
              19 December 2026
            </span>
          </p>
        </div>
        <div
          class="progress"
        >
          <div
            aria-label="25"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="25"
            class="progress-bar bg-info"
            role="progressbar"
            style="width: 25%"
          />
        </div>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - Project status planned renders correctly 1`] = `
<jest>
  <div
    class="bcl-project-status border-bottom pb-4 mb-4"
  >
    <div
      class="row"
    >
      <div
        class="col-12 col-md-3"
      >
        <p
          class="fw-bold"
        >
          Status
        </p>
        <span
          class="mb-3 mb-md-0 badge bg-secondary"
        >
          Not started
        </span>
      </div>
      <div
        class="col-12 col-md-9"
      >
        <div
          class="d-flex justify-content-between"
        >
          <p
            class="fw-bold"
          >
            Start
            <br />
            <span
              class="fw-normal"
            >
              19 December 2025
            </span>
          </p>
          <p
            class="fw-bold text-end"
          >
            End
            <br />
            <span
              class="fw-normal"
            >
              19 December 2027
            </span>
          </p>
        </div>
        <div
          class="progress"
        >
          <div
            aria-label="0"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="0"
            class="progress-bar bg-secondary"
            role="progressbar"
            style="width: 0%"
          />
        </div>
      </div>
    </div>
  </div>
</jest>
`;
